spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -303,13 +304,13 @@ public RetryTemplate buildRetryTemplate(ConsumerProperties properties) {
 		private static ConcurrentMap<String, MimeType> mimeTypesCache = new ConcurrentHashMap<>();
 
 		static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
+
 			Assert.notNull(payload, "payload object cannot be null.");
 			if (payload instanceof byte[]) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
-						: MimeTypeUtils.TEXT_PLAIN;
+				return originalContentType.toLowerCase().contains(MimeTypeUtils.APPLICATION_JSON.getSubtype()) ? MimeType.valueOf(originalContentType) : MimeTypeUtils.TEXT_PLAIN;
```

Comment:
@viniciusccarvalho Please check this set of changes https://github.com/ilayaperumalg/spring-cloud-stream/commit/a5b5bb2c4ca6188f4abfb5a15eff5bd3eef9e374 and let me know what is missing in terms of supporting the feauture this PR addresses. I am trying to understand why do we need to set the `contentType` while it is *only* used for `binder` serialization/de-serialization proces. 


>  If we modify it, then the contentType that was application/vnd.type.version+json will become application/json and then we loose the contentType on the flow.

As long as we keep the `originalContentType` I think we are still good retriveing the actual `contentType` because the `contentType` we set for serialization/de-serialization is only valid at the `binder` level.

> Although the originalContetType could preserve on the first hop but in my opinion is wrong what we do: contentType:text/plain, originalContentType:application/vnd.type.version+json.

here, I meant to use `contentType:application/json, originalContentType:application/vnd.type.version+json` during binder's serialization process.